api/lib/infrastructure/jobs/cpf-export/handlers/create-and-upload.js

```diff
@@ -1,20 +1,25 @@
 const { PassThrough } = require('stream');
+const bluebird = require('bluebird');
 
 module.exports = async function createAndUpload({
   data,
   cpfCertificationResultRepository,
   cpfCertificationXmlExportService,
   cpfExternalStorage,
 }) {
-  const { startId, endId } = data;
-  const cpfCertificationResults = await cpfCertificationResultRepository.findByIdRange({
-    startId,
-    endId,
+  const { complementaryCertificationIds } = data;
+  const cpfCertificationResults = await cpfCertificationResultRepository.findByCertificationCourseIds({
+    complementaryCertificationIds,
   });
 
   const writableStream = new PassThrough();
   cpfCertificationXmlExportService.buildXmlExport({ cpfCertificationResults, writableStream });
 
-  const filename = `pix-cpf-export-from-${startId}-to-${endId}.xml`;
+  const now = new Date().getTime();
+  const filename = `pix-cpf-export-${now}.xml`;
   await cpfExternalStorage.upload({ filename, writableStream });
+
+  await bluebird.each(cpfCertificationResults, async function ({ id: certificationCourseId }) {
+    await cpfCertificationResultRepository.markCertificationCourseAsExported({ certificationCourseId, filename });
+  });
```

Comment:
Il n'est pas nécessaire de multiplier les requêtes, on peut faire un whereIn(ids) plutôt qu'un where(id) dans le repo
```suggestion
    await cpfCertificationResultRepository.markCertificationCoursesAsExported({ certificationCourseIds, filename });

```

Reply:
Bien vu